azkaban-common/src/main/java/azkaban/project/ProjectManager.java

```diff
@@ -204,6 +204,34 @@ public Project getProject(int id) {
     return projectsById.get(id);
   }
 
+    /**
+     * fetch inactive project from db by project_id
+     *
+     * @param name
+     * @return
+     */
+    public Project getInactiveProject(String name) {
```

Comment:
function name is misleading, actually the function here will get a project with name match disregard the status, so it is not necessary to be Inactive.  Recommend to rename this to GetProject or GetProjectByName, state it in the meta that the returning result disregards the project status.  

Same comment for the function below, which gets the project by Id.